src/lib/db/competition.ts

```diff
@@ -126,3 +126,26 @@ export async function getLevelTypeFromCompetitionByCompetitionId(competitionId:
 
   return result.levelType;
 }
+
+export async function updateCompetitionById(competition: { 
+  id: number;
+  name: string;
+  location: string;
+  date: Date;
+  duration: number;
+  levelType: ClimbingGrade;
+  code: string;
+}) {
+  const { id, name, location, date, duration, levelType, code } = competition;
+  return await prisma.competition.update({
+    where: { id },
+    data: {
+      name,
+      location,
+      date,
+      duration,
+      levelType,
+      code,
+    },
+  });
+}
```

Comment:
```suggestion
export async function updateCompetitionById(competitionId: number, competitionData: Prisma.CompetitionUpdateInput) {
  return await prisma.competition.update({
    where: { id: competitionId },
    data: competitionData,
  });
}
```
Esto es más limpio y si en algun minuto se actualiza la information que tiene una competition no hay que preocuparse de andar actualizando ese endpoint

Reply:
Toda la razón, me gusta de esa forma. Arreglado!